### ISSUE TITLE
No handling for errors thrown inside of `renderCell` when fetching data dynamically

### ISSUE THREAD

**Issue by albertorestifo** [interval/interval-node#19]
If you throw an error inside of `renderCell` for an `io.display.table`, the table will show "No records founds" and there is no trace of the actual error, making debugging very difficult.

EDIT:

This only happens in combination with `getData`:

```ts
    return io.display.table('I will fail', {
      getData: async () => {
        return {
          data: [{ foo: 'bar' }],
        };
      },
      columns: [
        {
          label: 'Foo',
          renderCell: () => {
            throw new Error('Bazinga!');
          },
        },
      ],
    });
```

**Comment by jacobmischka** [comment]
Could you share more about the code that's causing this exact scenario? I'm unable to reproduce the "No records found" result you mentioned, I'm seeing the error bubble up as expected:

![image](https://user-images.githubusercontent.com/3939997/208488300-06a04b7c-878c-472f-9929-8283fc7db846.png)

```ts
    error_in_render_cell: async () => {
      const data = [
        { firstName: 'Annie', lastName: 'Adams', email: 'aa@example.com' },
        { firstName: 'Bob', lastName: 'Bryant', email: 'bb@example.com' },
        { firstName: 'Carl', lastName: 'Cranston', email: 'cc@example.com' },
      ]

      await io.display.table('Table', {
        data,
        columns: [
          { label: 'First name', accessorKey: 'firstName' },
          { label: 'Last name', accessorKey: 'lastName' },
          {
            label: 'Email',
            renderCell: row => {
              throw new Error('Oops!');

              return {
                label: row.email,
                url: `mailto:${row.email}`
              };
            }
          }
        ],
      });
    },
```

I'm also seeing the same happen in a Page's `children`. Am I missing something?

Though I agree it would be better for the SDK to handle errors in cells individually and not crash the entire action/page, thanks for the suggestion there!

**Comment by albertorestifo** [comment]
I'm still able to reproduce it: it seems to only happen in conjunction with a `getData` async function to fetch the data:

```ts
    return io.display.table('I will fail', {
      getData: async () => {
        return {
          data: [{ foo: 'bar' }],
        };
      },
      columns: [
        {
          label: 'Foo',
          renderCell: () => {
            throw new Error('Bazinga!');
          },
        },
      ],
    });
```

Results it:
![CleanShot 2022-12-20 at 10 18 05](https://user-images.githubusercontent.com/1037048/208629991-5a1ec7c4-575f-48c2-bbc9-8b60613cdfff.png)

**Comment by jacobmischka** [comment]
Thank you for the additional info! We've been able to reproduce and a fix is in review now that will go out in the next SDK release. We'll write back here when that happens, thanks again!

**Comment by jacobmischka** [comment]
This should be fixed in [v0.36.0](https://interval.com/changelog/0-36-0), thanks for letting us know!
